### PR TITLE
[AF-2366] BC Maven Archetype Support

### PR DESCRIPTION
**Pull request by caponetto**
**Task**: [AF-2366](https://issues.redhat.com/browse/AF-2366)

**Related PRs**:
- https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1151
- https://github.com/kiegroup/kie-soup/pull/124
- https://github.com/kiegroup/appformer/pull/872
- https://github.com/kiegroup/kie-wb-common/pull/3098
- https://github.com/kiegroup/jbpm-wb/pull/1406
- https://github.com/kiegroup/drools-wb/pull/1278
- https://github.com/kiegroup/optaplanner-wb/pull/352
- https://github.com/kiegroup/kie-wb-distributions/pull/1005

**DEMO**
AF-2366

**Notes**
The maven archetype used on the demo is available at [this link](https://drive.google.com/open?id=1k-7SJPtGIXc8-lK74Q3oZKgpgPGDfwfM).
Unzip it and execute `mvn clean install`, then the archetype will be available to be added into BC.

### CODE REVIEW

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by sthundat** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
:green_heart:

**Comment by ederign** [comment]
hi @sthundat can you please take a look at this? Do you have any estimative to finish the review?

**Comment by caponetto** [comment]
I had to rebase the branches (see https://github.com/kiegroup/kie-wb-common/pull/3098/commits/aa0748bc3d2f6464b5769538944796b34d66f042)

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
I had to rebase the branches once again.

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
Jenkins please retest this

File: uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/POMServiceImpl.java [hunk 1]
```diff
@@ -58,6 +58,8 @@
 public class POMServiceImpl
         implements POMService {
 
+    public static final String POM_XML = "pom.xml";
```
Comment (sthundat) [inline uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/POMServiceImpl.java]: looks like unused variable POM_XML

Comment (caponetto) [reply uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/POMServiceImpl.java]: It is being used in WorkspaceProjectServiceImpl.java, lines 327 and 328.

Comment (sthundat) [reply uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/POMServiceImpl.java]: the literal "pom.xml" is using on same file another places also.Please use the constant there as well

Comment (caponetto) [reply uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/POMServiceImpl.java]: Changing code where I originally haven't touched could lead to unwanted conflicts, but I've applied the changes you've requested. Please check my last commit.

**Comment by sthundat** [comment]
jenkins execute full downstream build

**Comment by sthundat** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
@paulovmr
The two failed tests are not related to this PR. 
Probably they will be fixed once https://github.com/kiegroup/jbpm-work-items/pull/120 is merged.

**Comment by caponetto** [comment]
I had to rebase the branches.

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
jenkins execute full downstream build

**Comment by caponetto** [comment]
All green!
@ederign @paulovmr

**Pull request merged**